app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -175,7 +178,16 @@ public void onCreate(Bundle savedInstanceState) {
                 //So we're not in landscape mode anymore, but were before. If we had something selected, we 
                 //need to go to the detail screen instead.
                 if(oldActivity != null) {
-                    if(oldActivity.selectedIntent != null) {
+                    Intent intent = this.getIntent();
+
+                    TreeReference selectedRef = SerializationUtil.deserializeFromIntent(intent,
+                            EntityDetailActivity.CONTEXT_REFERENCE, TreeReference.class);
+                    if (selectedRef != null) {
+                        Intent detailIntent = getDetailIntent(selectedRef, null);
+                        startOther = true;
+                        startActivityForResult(detailIntent, CONFIRM_SELECT);

```

Comment:
Has the bug where rotating from landscape to portrait while nothing is selected, but has been selected in the past, then that previous selection detail view is launched. Due to the fact that I don't know how to reset the CONTEXT_REFERENCE extra, I think...